src/back/spv/writer.rs

```diff
@@ -10,38 +11,19 @@ enum Signedness {
     Signed = 1,
 }
 
-#[derive(Debug)]
+#[derive(Hash, Eq, PartialEq, Debug)]
 enum LocalType {
     Scalar(crate::TypeInner),
```

Comment:
why are we storing `TypeInner` here? I.e. if we are going to store `TypeInner`, we wouldn't need `LocalType` at all :) The idea was to have these types fully specified, i.e.
```rust
enum LocalType {
  Scalar { kind, width },
  Vector { size, kind, width },
  Pointer { base, class },
}
```